app/controllers/notifications_controller.rb

```diff
@@ -0,0 +1,15 @@
+class NotificationsController < ApplicationController
+  def index
+    @notifications = current_user.notifications
+  end
+
+  def update
+    @notification = Notification.find(params[:id])
+
+    if @notifcation.mark_as_read!
+      redirect_to @notification.url
+    else
+      render :index
```

Comment:
What do you think about adding a flash here maybe something like:

`render :index,  notice: "Could not redirect to that notification"`